crates/cxx-qt-lib/src/core/qstringlist.rs

```diff
@@ -180,17 +193,30 @@ impl From<&QStringList> for QList<QString> {
     }
 }
 
+#[cfg(cxxqt_qt_version_at_least_6)]
```

Comment:
ugh it's unfortunate that Qt 5 seems different enough that the casting doesn't work.

Wonder how annoying it becomes to not be able to at least deref to a QList<QString> as yes then this is what is causing things like .append / .get to fail in the tests (and there are further tests too) that will only work with Qt 6.

So maybe that points that this would be painful without deref ability and you should bring back the old behaviour but only for Qt 5. So you need to use `cfg(cxxqt_qt_version_major = "5")` on the Rust side and `#if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))` on the C++ side